src/files/domain/models/File.ts

```diff
@@ -1,207 +1,6 @@
-  constructor(
-    readonly value: number,
-    readonly unit: FileSizeUnit,
-    readonly mode: FileDownloadMode
-  ) {
-    super(value, unit)
-    ;[this.value, this.unit] = FileDownloadSize.convertToLargestUnit(value, unit)
-  }
-}
-
-export interface FileAccess {
-  restricted: boolean
-  latestVersionRestricted: boolean
-  canBeRequested: boolean
-  requested: boolean
-}
-
-export enum FilePublishingStatus {
-  DRAFT = 'draft',
-  RELEASED = 'released',
-  DEACCESSIONED = 'deaccessioned'
-}
-
-export interface FileVersion {
-  number: number
-  publishingStatus: FilePublishingStatus
-}
-
-export enum FileDateType {
-  METADATA_RELEASED = 'metadataReleased',
-  PUBLISHED = 'published',
-  DEPOSITED = 'deposited'
-}
-
-export enum FileVersionNotNumber {
-  LATEST = 'latest',
-  DRAFT = 'draft'
-}
-
-export interface FileDate {
-  type: FileDateType
-  date: Date
-}
-
-export class FileEmbargo {
-  constructor(readonly dateAvailable: Date) {}
-
-  get isActive(): boolean {
-    return this.dateAvailable > new Date()
-  }
-}
-
-export interface FileTabularData {
-  variablesCount: number
-  observationsCount: number
-  unf?: string
-}
-
-export enum FileLabelType {
-  CATEGORY = 'category',
-  TAG = 'tag'
-}
-
-export interface FileLabel {
-  type: FileLabelType
-  value: string
-}
-
-export class FileType {
-  constructor(readonly value: string, readonly original?: string) {}
-
-  toDisplayFormat(): string {
-    return FileTypeToFriendlyTypeMap[this.value] || FileTypeToFriendlyTypeMap.unknown
-  }
-}
-
-export interface FileChecksum {
-  algorithm: string
-  value: string
-}
-
-export enum FileIngestStatus {
-  NONE = 'none',
-  IN_PROGRESS = 'inProgress',
-  SCHEDULED = 'scheduled',
-  ERROR = 'error'
-}
-
-export interface FileIngest {
-  status: FileIngestStatus
-  reportMessage?: string
-}
-
-export interface FileDownloadUrls {
-  original: string
-  tabular?: string
-  rData?: string
-}
-
-export class File {
-  constructor(
-    readonly id: number,
-    readonly version: FileVersion,
-    readonly name: string,
-    readonly access: FileAccess,
-    readonly type: FileType,
-    readonly size: FileSize,
-    readonly date: FileDate,
-    readonly downloadCount: number,
-    readonly labels: FileLabel[],
-    public readonly isDeleted: boolean,
-    public readonly ingest: FileIngest,
-    public readonly downloadUrls: FileDownloadUrls,
-    public thumbnail?: string,
-    readonly directory?: string,
-    readonly embargo?: FileEmbargo,
-    readonly tabularData?: FileTabularData,
-    readonly description?: string,
-    readonly checksum?: FileChecksum
-  ) {}
-
-  getLink(): string {
-    return `/file?id=${this.id}&version=${this.version.number}`
-  }
-
-  get isActivelyEmbargoed(): boolean {
-    if (this.embargo) {
-      return this.embargo.isActive
-    }
-    return false
-  }
-
-  get isTabularData(): boolean {
-    return this.tabularData !== undefined
-  }
+export interface File {
```

Comment:
It is a bit confusing that the class with the largest number of properties is the "preview", while "File" is only an interface with two properties.

Reply:
I see your point, but the File model is not finished yet, and when the File Page is finished the File model will contain all the FilePreview properties plus the file versions, plus the file metrics, etc. So at the end the File model will be larger than the File Preview, even though right now it's smaller because we only have the boilerplate properties.

Also, the way I see it is that in the Files Table of the Dataset View we're displaying the file information preview, while in the File View we're seeing the actual file.

Does it make sense to you?

Reply:
Makes sense. Thanks!